app/controllers/reports_controller.rb

```diff
@@ -31,6 +32,24 @@ def transfer_list
     report_responder('Transfer', current_organization, @transfers)
   end
 
+  def all_list
+    filename = "#{current_organization.name.gsub(' ', '_')}.zip"
+    temp_file = Tempfile.new(filename)
+    begin
+      Zip::OutputStream.open(temp_file) { |zos| }
+      Zip::File.open(temp_file.path, Zip::File::CREATE) do |zipfile|
+        add_csvs_to_zip(%w[Member Transfer Inquiries Offers], zipfile)
+      end
+      zip_data = File.read(temp_file.path)
+      send_data(zip_data, type: 'application/zip', disposition: 'attachment', filename: filename)
+    rescue Errno::ENOENT
+      redirect_to all_list_report_path
```

Comment:
This block is here because if you try to download the zip repeatedly without waiting, this error may appear, so a second attempt solves it.